src/config.rs

```diff
@@ -1,19 +1,48 @@
 //! All sync client configuration structs and logic
 use std::{
-	fmt::Display,
-	path::{Path, PathBuf},
+	ops::{Deref, DerefMut},
+	path::Path,
 };
 
 use anyhow::{bail, Result};
-use ldap_poller::{config::TLSConfig, AttributeConfig, CacheMethod, ConnectionConfig, Searches};
 use serde::Deserialize;
 use url::Url;
 
+use crate::{
+	sources::{csv::CsvSourceConfig, ldap::LdapSourceConfig, ukt::UktSourceConfig},
+	zitadel::ZitadelConfig,
+};
+
 /// App prefix for env var configuration
-const ENV_VAR_CONFIG_PREFIX: &str = "FAMEDLY_LDAP_SYNC";
+const ENV_VAR_CONFIG_PREFIX: &str = "FAMEDLY_SYNC";
 /// Separator for setting a list using env vars
 const ENV_VAR_LIST_SEP: &str = " ";
 
+/// The main sync tool with all configurations
+#[derive(Debug, Clone, Deserialize, PartialEq)]
+pub struct Config {
+	/// Configuration related to Zitadel provided by Famedly
+	pub zitadel_config: ZitadelConfig,
+	/// Sources configuration
+	pub sources_config: SourcesConfig,
+	/// Optional sync tool log level
+	pub log_level: Option<String>,
+	/// Opt-in features
+	#[serde(default)]
+	pub feature_flags: FeatureFlags,
+}
+
+/// Configuration for sources
+#[derive(Debug, Clone, Deserialize, PartialEq)]
+pub struct SourcesConfig {
+	/// Optional LDAP configuration
+	pub ldap_config: Option<LdapSourceConfig>,
+	/// Optional UKT configuration
+	pub ukt_config: Option<UktSourceConfig>,
+	/// Optional CSV configuration
+	pub csv_config: Option<CsvSourceConfig>,
+}
```

Comment:
Well, this is a product decision, as in: shall we implement the possibility for multiple UKT's, Ldap's, CSV's into the program?

Then it would be good to go back to making the `new` method part of the Source trait again, so that we can iterate over the vector of source configs and create each automatically. And adjust all the code accordingly.

While I appreciate your concerns, this is going back and forth like brainstorming way over the scope of what needed to be done here. Do we have to keep pushing all these ideas and possibilities into one PR? If deemed important, I suggest the way to go at this point would be to start creating new tasks for possible future extensions/adjustments/improvements.

Reply:
I don't see how making method `new` part of the trait would be neccessary or useful at all. We'd still need to pattern match on `AnySourceConfig` and call `new` depending on the variant, unless we do GADTs-style magic, which is a bit too much here especially in rust.

I'm merely pointing out weak points in initial requirements which I think are not that difficult to refine and implement in a better way from the start. But sure, we can factor it out into a different task and implement later

Reply:
I created a new issue then: https://github.com/famedly/ldap-sync/issues/57